### ISSUE TITLE
Install fails

### ISSUE THREAD

**Issue by K9PAL** [pybind/scikit_build_example#100]
Windows 11, Python 3.6, pip 21.3.1.

Repeat-by:
Follow instructions in README.md.
Git clone works. 
Then type pip install ./scikit_build_example

Expected: builds
Actual:
```
ERROR: Invalid requirement: './scikit_build_example'
```

Well, it's Windows, does changing the / to a \ help?
Expected: builds
Actual:
```
ERROR: Invalid requirement: '.\\scikit_build_example'
Hint: It looks like a path. File '.\scikit_build_example' does not exist.
What about changing it from . to .. so that there's something with the right name it can look at?
```
It also helps scikit-build-core. :) I'd say it's a hacky interm solution for all backends to pick up on setuptools configuraiton, but it's a reasonable solution until we have a better cross compiling standard.

Expected: builds:
Actual:

```pytb
Processing c:\users\beryl\source\repos\scikit_build_example
  Installing build dependencies ... error
  ERROR: Command errored out with exit status 1:
   command: 'c:\program files\python36\python.exe' 'c:\Program Files\python36\lib\site-packages\pip' install --ignore-installed --no-user --prefix 'C:\Users\beryl\AppData\Local\Temp\pip-build-env-n4hg686l\overlay' --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- 'scikit-build-core>=0.3.3' pybind11
       cwd: None
  Complete output (2 lines):
  ERROR: Could not find a version that satisfies the requirement scikit-build-core>=0.3.3 (from versions: none)
  ERROR: No matching distribution found for scikit-build-core>=0.3.3
  ----------------------------------------
WARNING: Discarding file:///C:/Users/beryl/source/repos/scikit_build_example. Command errored out with exit status 1: 'c:\program files\python36\python.exe' 'c:\Program Files\python36\lib\site-packages\pip' install --ignore-installed --no-user --prefix 'C:\Users\beryl\AppData\Local\Temp\pip-build-env-n4hg686l\overlay' --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- 'scikit-build-core>=0.3.3' pybind11 Check the logs for full command output.
ERROR: Command errored out with exit status 1: 'c:\program files\python36\python.exe' 'c:\Program Files\python36\lib\site-packages\pip' install --ignore-installed --no-user --prefix 'C:\Users\beryl\AppData\Local\Temp\pip-build-env-n4hg686l\overlay' --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- 'scikit-build-core>=0.3.3' pybind11 Check the logs for full command output.
```

**Comment by K9PAL** [comment]
Hello, Henry, it will be a few hours before I get to your suggestions but it is important to me to send a thank-you as quick as your detailed reply. The information about Python versions is exactly what a C++ person unfamiliar with the Python world (me) needs.

**Comment by K9PAL** [comment]
OK, I used the Windows uninstaller to (try to) get rid of 3.6, and downloaded and ran the 64 bit installer for 3.11 without error.

I am much further along thanks to your help, but am still encountering install errors. 

It turned out that more than one environment variable needed to be updated by hand. Did I perhaps get one wrong? Brace yourself, you're talking to someone who doesn't even know whether PYTHONPATH is just for my own libraries or whether it's supposed to point to the distributed ones. 

(I know what a file system is, I know what a codec is, but what on Earth is a "file system codec"?).



> C:\Users\beryl\source\repos>set | grep PYTHON
> MY_PYTHON_VERSION=311
> PYTHONHOME=c:\Users\beryl\AppData\Local\Programs\Python\Python311
> PYTHONPATH=T:\TimeLike\work_tls\timelike2.0\trunk\bin\windows\msvc15\x64;t:\TimeLike\timelike2_0_custom_libraries;
> PYTHON_EXE=c:\Users\beryl\AppData\Local\Programs\Python\Python311\python.exe
> PYTHON_INCLUDE=c:\Users\beryl\AppData\Local\Programs\Python\Python311\include
> PYTHON_LIB=c:\Users\beryl\AppData\Local\Programs\Python\Python311\libs\python311
> 
> C:\Users\beryl\source\repos>which python
> /c/Users/beryl/AppData/Local/Programs/Python/Python311/python
> 
> C:\Users\beryl\source\repos>python --version
> Python 3.11.4
> 
> C:\Users\beryl\source\repos>pip install ./scikit_build_example
> Fatal Python error: Py_Initialize: unable to load the file system codec
> Traceback (most recent call last):
>   File "c:\Users\beryl\AppData\Local\Programs\Python\Python311\lib\encodings\__init__.py", line 85, in search_function
>   File "c:\Users\beryl\AppData\Local\Programs\Python\Python311\lib\encodings\__init__.py", line 64, in normalize_encoding
> AttributeError: 'str' object has no attribute 'isascii'

**Comment by henryiii** [comment]
In general, you should have as few environment variables set as possible, preferably no Python specific ones at all, and Python should be on your PATH, and that's it (I am much more familiar with Unix-like systems, though, so maybe Windows needs one or more of those, but I don't think it needs many). Setting `PYTHONPATH` specifically is very likely to eventually break something. The "distributed" ones should not be in PYTHONPATH. It's much, much better to use virtual environments and (editable) installs; things are much less likely to break, and are more obvious than lots of environment variables, and you could support more than one version of Python that way eventually if you wanted to.

In your case, I think `pip` is still running from Python 3.6, since "isascii" was added in 3.7. `python -m pip` will likely work, since you've got a modern version of Python.

FYI, on Windows, I think the recommendation is to use `py` instead of `python`.

**Comment by K9PAL** [comment]
Thank you, Henry. That kind of advice is not at all obvious to someone who's worked outside the Python space, which makes it valuable.

Annd, you can close this! Here's the latest run.
----------

> C:\Users\beryl\source\repos>python -m pip install ./scikit_build_example
> Processing c:\users\beryl\source\repos\scikit_build_example
>   Installing build dependencies ... done
>   Getting requirements to build wheel ... done
>   Installing backend dependencies ... done
>   Preparing metadata (pyproject.toml) ... done
> Building wheels for collected packages: scikit-build-example
>   Building wheel for scikit-build-example (pyproject.toml) ... done
>   Created wheel for scikit-build-example: filename=scikit_build_example-0.0.1-cp311-cp311-win_amd64.whl size=55232 sha256=fe3ff11ef211061cb0d6ccfe50f3eb442991b9774cb2640f7062798d7099b85f
>   Stored in directory: C:\Users\beryl\AppData\Local\Temp\pip-ephem-wheel-cache-bjft8p0v\wheels\e5\c2\03\519253345761c48802b32aac5e1dadcd05e75454ec9604e501
> Successfully built scikit-build-example
> Installing collected packages: scikit-build-example
> Successfully installed scikit-build-example-0.0.1
> 
> [notice] A new release of pip is available: 23.1.2 -> 23.2.1
> [notice] To update, run: python.exe -m pip install --upgrade pip
